api/v1alpha4/machinedeployment_webhook.go

```diff
@@ -45,6 +46,11 @@ var _ webhook.Validator = &MachineDeployment{}
 // Default implements webhook.Defaulter so a webhook will be registered for the type.
 func (m *MachineDeployment) Default() {
 	PopulateDefaultsMachineDeployment(m)
+	// tolerate version strings without a "v" prefix: prepend it if it's not there
+	if m.Spec.Template.Spec.Version != nil && !strings.HasPrefix(*m.Spec.Template.Spec.Version, "v") {
+		normalizedVersion := "v" + *m.Spec.Template.Spec.Version
```

Comment:
Doesn't this cause a rollout because we're still hashing the Template spec?

Reply:
Do you mean mutating the value will result in the resource being marked dirty and then a new reconciliation will kick off (in the middle of the existing reconciliation)?

Reply:
Yes

Reply:
Why doesn't the mutation in the properties handled by the `PopulateDefaultsMachineDeployment` func have the same undesirable outcome?

Reply:
It technically does, but the implementation of `PopulateDefaultsMachineDeployment` makes sure to only set certain things that are either empty, or forced (which are there from when the object is created.

Reply:
@vincepri wouldn't this be a one-time thing though, only if the user creates a MD without the v prefix (which as described in #4651 isn't functional today)? Once the template version has the v prefix it won't modify it / cause a rollout

Reply:
That makes sense, I was assuming that the version would go through regardless, but it seems that it doesn't.